modeshape-jcr/src/test/java/org/modeshape/jcr/RepositoryConfigurationTest.java

```diff
@@ -408,6 +409,28 @@ public void shouldReadJournalingConfiguration() {
         assertValid("config/repo-config-journaling.json");
     }
 
+    @Test
+    public void shouldAllowClusteringToBeConfigured() throws Exception {
+        String clusterName = "testCluster";
+        String channelConfig = "someConfig";
+
+        RepositoryConfiguration config = RepositoryConfiguration.read("{ \"clustering\" : {\"clusterName\":\"" + clusterName
+                                                                      + "\", \"configuration\": \"" + channelConfig
+                                                                      + "\"} }");
+        RepositoryConfiguration.Clustering clusteringConfiguration = config.getClustering();
+        assertEquals(clusterName, clusteringConfiguration.getClusterName());
+        assertEquals(channelConfig, clusteringConfiguration.getConfiguration());
+    }
+    
+    @Test
+    public void shouldUseDefaultClusteringValues() throws Exception {
+        RepositoryConfiguration config = RepositoryConfiguration.read("{ \"clustering\" : {} }");
+        RepositoryConfiguration.Clustering clusteringConfiguration = config.getClustering();
+        assertEquals(Default.CLUSTER_NAME, clusteringConfiguration.getClusterName());
+        assertEquals(Default.CLUSTER_CONFIG, clusteringConfiguration.getConfiguration());
+    }
```

Comment:
Would it be useful to have a test that verifies that no clustering is used when the clustering nested document is *missing*?